hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSTableDescriptors.java

```diff
@@ -94,6 +95,10 @@
   // TODO.
   private final Map<TableName, TableDescriptor> cache = new ConcurrentHashMap<>();
 
+  // Default meta table descriptor, will be used by RegionServer during rolling upgrade until
+  // HMaster write latest 2.x meta table descriptor
+  private TableDescriptor defaultMetaTableDesc = null;
```

Comment:
What happens when using this "default descriptor" the regionserver attempts to update meta? Can that happen? We are assuming the master will rewrite soon, but is that valid? Regionservers don't run masters, they rely on an operator to do that. Who knows what the operator is doing. 

Would it be better to address specific fallbacks where something is missing or not expected?

Reply:
RegionServer can't update the meta descriptor,
    org.apache.hadoop.hbase.regionserver.HRegionServer#canUpdateTableDescriptor